AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -52,9 +54,12 @@ - (instancetype)initWithAsyncDataFetching:(BOOL)asyncDataFetchingEnabled
     return nil;
   }
   
-  _completedNodes = [NSMutableArray array];
-  _editingNodes = [NSMutableArray array];
+  _completedNodes = [NSMutableDictionary dictionary];
```

Comment:
This is a pretty big change.  It's the only thing I've seen so far that definitely needs to be profiled.  We will almost surely land without changing it, but sometime in the future, we should consider moving to a C++ type or another solution faster than a large dictionary with main-thread-hot key access.